### PR TITLE
GPU fixes

### PR DESCRIPTION
**Pull request by victorapm**
- `hypre_CSRMatrixMergeColMapOffd`: Add implementation for corner cases failing with cuda 12.2.2
- Fix regression tests with umpire on lassen

### CODE REVIEW

**Review by rfalgout — APPROVED** [review]
Looks good to me, but @liruipeng is the expert.  Thanks!

File: AUTOTEST/machine-lassen.sh [hunk 1]
```diff
@@ -97,8 +97,8 @@ ro="-struct -rt -mpibind -save ${save}"
 ./renametest.sh basic $output_dir/basic-cuda-nonum
 
 # CUDA without UM with umpire [benchmark]
-UMPIRE_DIR=/usr/workspace/hypre/ext-libs/Umpire/2022.03.1-nvcc11.2-sm_70-xl2023.06.28-cuda-11.2.0-gcc-8.3.1
-co="--with-cuda --with-gpu-arch=70 --with-umpire --with-umpire-include=${UMPIRE_DIR}/include --with-umpire-lib-dirs=${UMPIRE_DIR}/lib --with-umpire-libs=umpire --with-extra-CFLAGS=\\'-qsuppress=1500-029\\' --with-extra-CXXFLAGS=\\'-qsuppress=1500-029\\'"
+UMPIRE_DIR=/usr/workspace/hypre/ext-libs/Umpire/install_umpire-2025.03.0_nvcc11.2-sm_70-xl2023.06.28-cuda-11.2.0-gcc-8.3.1
+co="--with-cuda --with-gpu-arch=70 --with-umpire --with-umpire-include=${UMPIRE_DIR}/include --with-umpire-lib-dirs=${UMPIRE_DIR}/lib --with-umpire-libs=\\'camp umpire\\' --with-extra-CFLAGS=\\'-qsuppress=1500-029\\' --with-extra-CXXFLAGS=\\'-qsuppress=1500-029\\'"
```
Comment (liruipeng) [inline AUTOTEST/machine-lassen.sh:101]: Thanks for the fix!

File: src/seq_mv/csr_matop_device.c [hunk 2]
```diff
@@ -436,6 +436,46 @@ hypre_CSRMatrixMergeColMapOffd( HYPRE_Int      num_cols_offd_B,
...
+   {
+      *num_cols_offd_C_ptr = 0;
+      *col_map_offd_C_ptr  = NULL;
+      *map_B_to_C_ptr      = NULL;
+
+      return hypre_error_flag;
+   }
+   else if (!col_map_offd_B && B_ext_offd_bigj)
+   {
+      *num_cols_offd_C_ptr = B_ext_offd_nnz;
```
Comment (liruipeng) [inline src/seq_mv/csr_matop_device.c]: I don't understand this part. Can we talk about it?

**Comment by victorapm** [comment]
[6ebc5b7](https://github.com/hypre-space/hypre/pull/1291/commits/6ebc5b7b9823f4d62b49af6374b4f1274a7063bc) refactors the new specialized code and make the implementation of `hypre_CSRMatrixMergeColMapOffd` cleaner. Thanks to @liruipeng!

**Review by liruipeng — APPROVED** [review]
Looks good now. Thanks!
